### DISCUSSION TITLE
Add migration guide from vanilla React

### DISCUSSION THREAD

**Discussion by leerob** [vercel/next.js#62362] [Help]
# Feature request

As a user of [x] framework/language, I'd love to see an easy path towards migrating to Next.js

## Describe the solution you'd like

Documentation around common providers and how exactly you can switch to Next.js from them. 

## Additional context

Let's start with a few of the most popular and see what others in the community would get value from. 

- Create React App (https://github.com/vercel/next.js/issues/17454)
- Gatsby (https://github.com/vercel/next.js/issues/17453)
- Vanilla React

**Comment by Maxgit3** [comment]
I would be interested in helping with a migration guide for Vanilla React.  Would it be a comparison of building an app in React vs Next js or a comparison of the features?  I have been working with React and Next js.  I am new to open source contribution.  I am willing to help anyway I can.

**Comment by leerob** [comment]
Hey @Maxgit3! Thanks for offering to help. I just removed "good first issue" because I don't want to lead you down the wrong path. The scope of what I have in mind for this will involve advanced React/Next knowledge. 

- If I've ejected Create React App (or have hand-rolled my own React setup with Webpack / Babel), how can I understand the differences in my custom setup versus what Next provides by default? Link to docs for Webpack and Babel overrides for Next.
- Try to find the most common reasons people eject Create React App and what those configurations are. Then, if possible, show how they can be accomplished with Next.
- There's probably an `index.html` file that needs to be converted into the Next way of doing things
- Link to the "Migrating from Create React App" documentation when it's finished to include how to transform using React Router to Next's file-system based routing
- If you have custom file loaders set up for CSS/Sass/assets, explain how that's all built into Next and show how to convert. 
- Talk about new features that are added by default in Next that you can probably remove if you had them added manually (optional chaining, polyfilling fetch, etc)
- If you had custom code splitting, you can remove that
- Probably wouldn't hurt to mention how env variables are handled with next (specifically thinking how you prefix with `NEXT_PUBLIC_` to have it exposed client side)

**Comment by Maxgit3** [comment]
Thanks for the update

**Comment by gfortaine** [comment]
[Gradual Migration Strategies For Next.js](http://op.co.de/WfVUdF)

**Comment by samcx** [comment]
Moving this to <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [discussions](https://github.com/vercel/next.js/discussions) since this is a feature request!
